common/common-client/src/main/java/org/ow2/proactive/authentication/IAMLoginModule.java

```diff
@@ -0,0 +1,393 @@
+
+            // gets the username, password, group Membership, and group Hierarchy from callback handler
+            callbackHandler.handle(callbacks);
+            Map<String, Object> params = ((NoCallback) callbacks[0]).get();
+            String username = (String) params.get("username");
+            String password = (String) params.get("pw");
+
+            params.clear();
+            ((NoCallback) callbacks[0]).clear();
+
+            CommonUtils.assertNotNull(username, "No username has been specified for authentication");
+            CommonUtils.assertNotNull(password, "No password has been specified for authentication");
+
+            assertion = ticketValidator.validate(getServiceToken(username, password), service);
+
+            CommonUtils.assertNotNull(assertion,
+                                      "Empty assertion returned by ticket validator " + this.ticketValidatorClass);
+
+            String authenticator = assertion.getPrincipal().getName();
+            String role = (String) assertion.getPrincipal().getAttributes().get(roleAttributeName);
+
+            CommonUtils.assertNotNull(authenticator,
+                                      "Authenticator returned by ticket validator " + this.ticketValidatorClass +
+                                                     " is null");
+            CommonUtils.assertNotNull(role,
+                                      "User role returned by ticket validator " + this.ticketValidatorClass +
+                                            " is null");
+
+            //Authentication
+            subject.getPrincipals().add(new UserNamePrincipal(authenticator));
+            LOG.debug("adding principal '" + authenticator);
+
+            //Authorization
+            String[] groups = role.split(roleSeparator);
+            for (String group : groups) {
+                subject.getPrincipals().add(new GroupNamePrincipal(group.trim()));
+                LOG.debug("adding group '" + group + "' to principal '" + authenticator + "'");
+            }
+
+            //Successful login
+            succeeded = true;
+            LOG.debug("authentication succeeded for user '" + authenticator + "'");
+
+            return succeeded;
+
+        } catch (IOException | UnsupportedCallbackException | TicketValidationException e) {
+            LOG.error("", e);
+            throw new LoginException(e.toString());
+        }
+    }
+
+    /**
+     * <p>
+     * This method is called if the LoginContext's overall authentication
+     * succeeded (the relevant REQUIRED, REQUISITE, SUFFICIENT and OPTIONAL
+     * LoginModules succeeded).
+     * <p>
+     *
+     * @exception LoginException
+     *                if the commit fails.
+     *
+     * @return true if this IAMLoginModule's own login and commit attempts
+     *         succeeded, or false otherwise.
+     */
+    @Override
+    public boolean commit() throws LoginException {
```

Comment:
I think that the doc needs to be updated according to the method body.

Reply:
Done